src/servers/http_server.cc

```diff
@@ -296,54 +304,76 @@ HTTPAPIServer::ResponseAlloc(
     const char* tensor_name, size_t byte_size,
     TRTSERVER_Memory_Type memory_type, int64_t memory_type_id, void* userp)
 {
-  evbuffer* evhttp_buffer = reinterpret_cast<evbuffer*>(userp);
+  auto userp_pair = reinterpret_cast<EVBufferPair*>(userp);
+  evbuffer* evhttp_buffer = reinterpret_cast<evbuffer*>(userp_pair->first);
+  const std::unordered_map<std::string, std::pair<const void*, size_t>>&
+      output_shm_map = userp_pair->second;
 
   *buffer = nullptr;
   *buffer_userp = nullptr;
 
-  // Don't need to do anything if no memory was requested.
-  if (byte_size > 0) {
-    // Can't allocate for any memory type other than CPU.
-    if (memory_type != TRTSERVER_MEMORY_CPU) {
-      LOG_VERBOSE(1) << "HTTP allocation failed for type " << memory_type
-                    << " for " << tensor_name;
-      return nullptr;  // Success
-    }
-    
-    // Reserve requested space in evbuffer...
-    struct evbuffer_iovec output_iovec;
-    if (evbuffer_reserve_space(evhttp_buffer, byte_size, &output_iovec, 1) !=
-        1) {
-      return TRTSERVER_ErrorNew(
-          TRTSERVER_ERROR_INTERNAL,
-          std::string(
-              "failed to reserve " + std::to_string(byte_size) +
-              " bytes in output tensor buffer")
-              .c_str());
-    }
+  // Can't allocate for any memory type other than CPU.
+  if (memory_type != TRTSERVER_MEMORY_CPU) {
+    LOG_VERBOSE(1) << "HTTP allocation failed for type " << memory_type
+                   << " for " << tensor_name;
+    return nullptr;  // Success
+  }
 
-    if (output_iovec.iov_len < byte_size) {
-      return TRTSERVER_ErrorNew(
-          TRTSERVER_ERROR_INTERNAL,
-          std::string(
-              "reserved " + std::to_string(output_iovec.iov_len) +
-              " bytes in output tensor buffer, need " +
-              std::to_string(byte_size))
-              .c_str());
+
+  if (!output_shm_map.empty()) {
```

Comment:
This logic isn't right. If some output is in map, but not 'tensor_name' then we want to use existing evbuffer logic for the output. Just use find() and only if you find the output do you use shared memory.

Reply:
Fixed